analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -46,7 +46,7 @@ public RoslynSymbolicExecution(ControlFlowGraph cfg, SymbolicCheck[] checks, Can
             {
                 throw new ArgumentException("At least one check is expected", nameof(checks));
             }
-            this.checks = new(new SymbolicCheck[] { new ConstantCheck() }.Concat(checks).ToArray());
+            this.checks = new(new SymbolicCheck[] { new NotNullCheck(), new ConstantCheck() }.Concat(checks).ToArray());
```

Comment:
The `NotNullCheck` sounds a bit broad for all the things the engine does. Can it be `NonNullableValueTypeCheck` to keep it's scope very limited?